### ISSUE TITLE
[Nice to have] Multi Window Support

### ISSUE THREAD

**Issue by russkyc** [DotNetExtension/BlazorDesktop#42]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

No

### Describe the solution you'd like

Currently the BlazorDesktop platform runs on a single window, understandably so. For some advanced cases, it might be useful to have multi window support. Maybe having an extension to the `NavigationManager` or having something like an `IWindowManager` service where it will also handle the main window as well as managing other windows, this might be a great addition in the future.

```razor
@inject NavigationManager Navigation
@inject IWindowManager WindowManager

@code
{

    void OnOpenInNewWindow()
    {
        Navigation.NavigateToInNewWindow("/products");
    }

    void OnOpenNewWindow()
    {
        WindowManager.Create("window-identifier");
        var window = WindowManager.GetWindow("window-identifier");
        window.NavigateTo("/products");
    }
    
}
```

**Comment by andrewbabbittdev** [comment]
This has been something I have wanted to do for a while, and have known there would be some who want or need this. I just haven't had the time / need to work on it yet. I do have a project using this that could see some benefit from this, however its just not needed at this current moment.

Most of the work & patches lately have been focused on bug fixes and gaps from initial releases.

**Comment by russkyc** [comment]
Happy to say that everything else have been smooth, I'll try to play around and see if I can do a contribution for this if time allows. Thanks for working on this package!

**Comment by pv-parimal** [comment]
@andrewbabbittdev could you provide pointers on how to achieve this, I can give it a shot ?

**Comment by andrewbabbittdev** [comment]
I had some ideas over the past week around making a new abstract Blazor Window component that wraps around the native WPF window.

Main window would then be registered under the host builder and we could expose many window properties through there. This is just a rough idea of what I have been thinking about at least

**Comment by andrewbabbittdev** [comment]
@russkyc @pv-parimal 

I'd like to get this implemented in the BlazorDesktop 10.0 release.

I'd like to start taking API proposals to figure out how this feature should actually function before we start any coding work as there are quite a few ways this could be implemented and each have their own pros and cons.

On top of implementing this feature, I would also like for the proposed solution to address #55 and #65

The proposal in this issue is missing things, in particular:

- How are windows registered (thinking beyond this single hello world example, what happens if i register the same window in 2 places). What about using appsettings? How do we transition from the existing config model?
- How is window state managed, id assume the service container?
- How would an extension method on `NavigationManager` even access the IServiceProvider needed to access the windows / window state?

Its good on the surface level, but it needs more fleshing out before we can do anything.

I have also thought about what if we use the component model for windows?

Example:

MainWindow.razor
```razor
<Window Height="200" Width="500" Center="true" />
```

Program.cs
```cs
var builder = BlazorDesktopHostBuilder.CreateDefault(args);

builder.RootComponents.Add<BlazorDesktop.Sample.Components.Routes>("#app");
builder.RootComponents.Add<HeadOutlet>("head::after");

if (builder.HostEnvironment.IsDevelopment())
{
    builder.UseDeveloperTools();
}

builder.AddWindows<App>();

await builder.Build().RunAsync();
```

Potential benefits include the ability to control window visibility using standard razor logic:

```razor
@if (someFlag == true)
{
    <SettingsWindow />
}
```

I still haven't fully through over how this would actually work under the hood, but its some initial thinking and idk what people prefer from a consumption perspective.

**Comment by 0xApp** [comment]
@andrewbabbittdev  I have raised a PR for this multi-window implementation. https://github.com/DotNetExtension/BlazorDesktop/pull/72
